ui/src/path.ts

```diff
@@ -37,24 +32,41 @@ export const projectIssues = (id: string): string => `/projects/${id}/issues`;
 export const projectIssue = (id: string): string => `/projects/${id}/issue`;
 export const projectRevisions = (id: string): string =>
   `/projects/${id}/revisions`;
+
 export const projectSource = (
-  id: string,
-  revision: string,
-  objectType: string,
-  path: string
+  projectId: string,
+  userId: string,
+  revision = "",
+  objectType: string = ObjectType.Tree,
+  objectPath = ""
 ): string => {
-  if (revision && path) {
-    return `/projects/${id}/source/${revision}/${objectType}/${
-      objectType === ObjectType.Tree ? `${path}/` : path
-    }`;
-  } else {
-    return `/projects/${id}/source`;
-  }
+  return (
+    `/projects/${projectId}/${userId}/source?` +
+    `revision=${revision}&` +
+    `objectType=${objectType}&` +
+    `objectPath=${objectPath}`
+  );
+};
+
+export const parseProjectSourceLocation = (
+  querystring: string,
+  defaultRevision: string
+): any => {
+  const parsed = parse(querystring);
+  return {
+    currentRevision: parsed.revision || defaultRevision,
+    currentObjectType: parsed.objectType,
+    currentObjectPath: parsed.objectPath,
+  };
 };
-export const projectCommit = (id: string, hash: string): string =>
-  `/projects/${id}/commit/${hash}`;
-export const projectCommits = (id: string, branch: string): string =>
-  `/projects/${id}/commits/${encodeURIComponent(branch)}`;
+
+export const projectCommit = (id: string, user: string, hash: string): string =>
+  `/projects/${id}/${user}/commit/${hash}`;
+export const projectCommits = (
+  id: string,
+  user: string,
```

Comment:
Let's figure out what this "user" is: If it's a URN, eg. `rad:git:12389asd8`, then `user` is appropriate. If it's a PeerID on the other hand, we should call this `peer`.

Reply:
I think `userUrn` would be more helpful. `user: String` doesn't portray much to me, it could just be the handle for all I know (*cough* types *cough*). So I think `userUrn` or `peerId` would more indicative.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/620/commits/d5b10c171dddbf5b93c4a3b1493d25c1689799f0